### PR TITLE
fix #16526 run config.nims before foo.nim.cfg

### LINKED ISSUE #16526: config file resolution might be broken or reported incorrectly
Config file resolution for Nimscript evaluation is either incorrect, misreported, or surprising if as it should be.

The resolution of files reports on the command line from least to most specific. When evaluating a Nimscript (project.nims) the resolution considered the config.nims more specific than a project.nim.cfg. This seems surprising.

### Example
```shell
$ touch foo.nim
$ touch foo.nims
$ touch foo.nim.cfg
$ touch config.nims
$ touch nim.cfg
$ nim e foo.nims
```

### Current Output
```
Hint: used config file '/home/saem/Development/nim/Nim/config/nim.cfg' [Conf]
Hint: used config file '/home/saem/Development/nim/Nim/config/config.nims' [Conf]
Hint: used config file '/home/saem/Development/nim/nim-misc/config-loading-issue/nim.cfg' [Conf]
Hint: used config file '/home/saem/Development/nim/nim-misc/config-loading-issue/foo.nim.cfg' [Conf]
Hint: used config file '/home/saem/Development/nim/nim-misc/config-loading-issue/config.nims' [Conf]
Hint: used config file '/home/saem/Development/nim/nim-misc/config-loading-issue/foo.nims' [Conf]
```

### Expected Output
```
Hint: used config file '/home/saem/Development/nim/Nim/config/nim.cfg' [Conf]
Hint: used config file '/home/saem/Development/nim/Nim/config/config.nims' [Conf]
Hint: used config file '/home/saem/Development/nim/nim-misc/config-loading-issue/nim.cfg' [Conf]
Hint: used config file '/home/saem/Development/nim/nim-misc/config-loading-issue/config.nims' [Conf]
Hint: used config file '/home/saem/Development/nim/nim-misc/config-loading-issue/foo.nim.cfg' [Conf]
Hint: used config file '/home/saem/Development/nim/nim-misc/config-loading-issue/foo.nims' [Conf]
```

### Possible Solution

* prioritize module/project name match over kind (nimscript) match

```
$ nim --version
Nim Compiler Version 1.5.1 [Linux: amd64]
Compiled at 2020-12-31
Copyright (c) 2006-2020 by Andreas Rumpf

git hash: 876fa3e62e41cd366b89137cc3c4f6b5b8b2bee8
active boot switches:
```

### PR DESCRIPTION
**Pull request by timotheecour**
fix #16526

### CODE REVIEW

File: tests/misc/trunner.nim [hunk 1]
```diff
@@ -224,6 +224,25 @@ sub/mmain.idx""", context
...
+tests/config.nims
+tests/newconfig/bar/nim.cfg
+tests/newconfig/bar/config.nims
+tests/newconfig/bar/mfoo.nim.cfg
+tests/newconfig/bar/mfoo.nims""".splitLines
+    var expected = ""
+    for a in files:
+      let b = dir / a
+      expected.add &"Hint: used config file '{b}' [Conf]\n"
+    doAssert outp == expected, outp & "\n" & expected
```
Comment (Clyybber) [inline tests/misc/trunner.nim:244]: Can't testament spec be used for this?

Comment (timotheecour) [reply tests/misc/trunner.nim:244]: how? the problem is `cwd`

Comment (Clyybber) [reply tests/misc/trunner.nim:244]: But there is `listFullPaths:off`

Comment (Clyybber) [reply tests/misc/trunner.nim:244]: Nevermind, seems like that has no effect on these messages.

**Pull request merged**